### ISSUE TITLE
錯誤訊息

### ISSUE THREAD

**Issue by dogodg3838** [EvilBT/ComfyUI_SLK_joy_caption_two#179]
<img width="605" height="612" alt="Image" src="https://github.com/user-attachments/assets/6e4f855d-d439-456c-8189-f6ce7cb7bc4a" />

錯誤資訊
# ComfyUI Error Report
## Error Details
- **Node ID:** 70
- **Node Type:** Joy_caption_two
- **Exception Type:** TypeError
- **Exception Message:** 'NoneType' object is not subscriptable

## Stack Trace
```
  File "D:\Pinokio\api\comfy.git\app\execution.py", line 496, in execute
    output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)

  File "D:\Pinokio\api\comfy.git\app\execution.py", line 315, in get_output_data
    return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)

  File "D:\Pinokio\api\comfy.git\app\custom_nodes\comfyui-lora-manager\py\metadata_collector\metadata_hook.py", line 165, in async_map_node_over_list_with_metadata
    results = await original_map_node_over_list(

  File "D:\Pinokio\api\comfy.git\app\execution.py", line 289, in _async_map_node_over_list
    await process_inputs(input_dict, i)

  File "D:\Pinokio\api\comfy.git\app\execution.py", line 277, in process_inputs
    result = f(**inputs)

  File "D:\Pinokio\api\comfy.git\app\custom_nodes\comfyui_slk_joy_caption_two\joy_caption_two_node.py", line 342, in generate
    embedded_images = joy_two_pipeline.image_adapter.embedded_image(vision_outputs.hidden_states)

  File "D:\Pinokio\api\comfy.git\app\custom_nodes\comfyui_slk_joy_caption_two\joy_caption_two_node.py", line 144, in embedded_image
    embedded_images = self.image_adapter(hidden_states)

  File "D:\Pinokio\api\comfy.git\app\env\lib\site-packages\torch\nn\modules\module.py", line 1739, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)

  File "D:\Pinokio\api\comfy.git\app\env\lib\site-packages\torch\nn\modules\module.py", line 1750, in _call_impl
    return forward_call(*args, **kwargs)

  File "D:\Pinokio\api\comfy.git\app\custom_nodes\comfyui_slk_joy_caption_two\joy_caption_two_node.py", line 101, in forward
    x = vision_outputs[-2]

**Comment by wzy79045179-del** [comment]
一样的错误

**Comment by finger92** [comment]
This should be a version issue with the python library. Have you ever tried "pip install - r rquirements. txt"

**Comment by p245986-lab** [comment]
我也碰到这个问题了，重新安装python库也没有解决

**Comment by finger92** [comment]
这是我这里的pip list，可以参考一下（工作流比较多所以依赖也比较多）

```
absl-py                        2.3.1
accelerate                     1.10.1
addict                         2.4.0
aiofiles                       24.1.0
aiohappyeyeballs               2.6.1
aiohttp                        3.12.15
aiohttp_socks                  0.10.1
aiosignal                      1.4.0
albucore                       0.0.24
albumentations                 2.0.8
alembic                        1.16.5
annotated-types                0.7.0
antlr4-python3-runtime         4.9.3
anyio                          4.11.0
archspec                       0.2.5
asttokens                      3.0.0
astunparse                     1.6.3
attrs                          25.3.0
audioread                      3.0.1
audiotsm                       0.1.2
av                             15.1.0
backports.tarfile              1.2.0
BasicLibrary.PY                0.5.39
beautifulsoup4                 4.13.4
bitsandbytes                   0.47.0
blend_modes                    2.2.0
boltons                        25.0.0
boto3                          1.40.51
botocore                       1.40.51
Brotli                         1.1.0
build                          1.3.0
CacheControl                   0.14.3
cachetools                     6.2.0
certifi                        2025.8.3
cffi                           2.0.0
cfgv                           3.4.0
chardet                        5.2.0
charset-normalizer             3.4.2
cleo                           2.1.0
click                          8.2.1
clip-interrogator              0.6.0
cmake                          4.0.3
cn2an                          0.5.23
color-matcher                  0.6.0
colorama                       0.4.6
coloredlogs                    15.0.1
colorlog                       6.9.0
colour-science                 0.4.6
comfyui-embedded-docs          0.3.0
comfyui_frontend_package       1.27.10
comfyui_workflow_templates     0.1.95
comtypes                       1.4.12
conda                          25.7.0
conda-build                    25.7.0
conda_index                    0.6.1
conda-libmamba-solver          25.3.0
conda-package-handling         2.4.0
conda_package_streaming        0.12.0
contourpy                      1.3.3
crashtest                      0.4.1
cryptography                   46.0.1
cssselect2                     0.8.0
cstr                           0.1.0
ctranslate2                    4.6.0
cycler                         0.12.1
Cython                         3.1.4
datasets                       4.2.0
ddt                            1.7.2
decorator                      5.2.1
decord                         0.6.0
diffusers                      0.35.1
dill                           0.4.0
diskcache                      5.6.3
distlib                        0.4.0
distro                         1.9.0
dnspython                      2.7.0
docstring_parser               0.17.0
docutils                       0.22.2
dtw-python                     1.7.2
dulwich                        0.24.2
easydict                       1.13
edge-tts                       7.2.3
einops                         0.8.1
einx                           0.3.0
embreex                        2.17.7.post6
encodec                        0.1.1
eng-to-ipa                     0.0.2
evalidate                      2.0.5
exceptiongroup                 1.3.0
executing                      2.2.0
expecttest                     0.3.0
face-alignment                 1.4.1
facexlib                       0.3.0
fairscale                      0.4.13
fal_client                     0.7.0
fastapi                        0.119.0
faster-whisper                 1.2.0
fastjsonschema                 2.21.2
ffmpy                          0.3.0
filelock                       3.18.0
filterpy                       1.4.5
findpython                     0.7.0
fire                           0.5.0
flatbuffers                    25.9.23
fonttools                      4.60.0
freetype-py                    2.5.1
frozendict                     2.4.6
frozenlist                     1.7.0
fsspec                         2025.7.0
ftfy                           6.3.1
func-timeout                   4.3.5
future                         1.0.0
fvcore                         0.1.5.post20221221
gast                           0.6.0
gdown                          5.2.0
gguf                           0.17.1
gitdb                          4.0.12
GitPython                      3.1.45
google-pasta                   0.2.0
gradio                         5.49.1
gradio_client                  1.13.3
greenlet                       3.2.4
groovy                         0.1.2
grpcio                         1.75.1
h11                            0.16.0
h2                             4.2.0
h5py                           3.14.0
hf-xet                         1.1.10
hpack                          4.1.0
httpcore                       1.0.9
httpx                          0.28.1
httpx-sse                      0.4.1
huggingface-hub                0.23.4
humanfriendly                  10.0
hydra-core                     1.3.2
hyperframe                     6.1.0
hypothesis                     6.137.1
identify                       2.6.14
idna                           3.10
imageio                        2.37.0
imageio-ffmpeg                 0.6.0
img2texture                    1.0.6
importlib_metadata             8.7.0
importlib_resources            6.5.2
inflect                        7.5.0
iniconfig                      2.1.0
insightface                    0.7.3
installer                      0.7.0
instructor                     1.11.3
iopath                         0.1.10
ipython                        9.4.0
ipython_pygments_lexers        1.1.1
jaraco.classes                 3.4.0
jaraco.context                 6.0.1
jaraco.functools               4.3.0
jax                            0.8.0
jaxlib                         0.8.0
jedi                           0.19.2
jeepney                        0.9.0
jieba                          0.42.1
Jinja2                         3.1.6
jiter                          0.10.0
jmespath                       1.0.1
joblib                         1.5.2
json_repair                    0.51.0
jsonpatch                      1.33
jsonpointer                    3.0.0
jsonschema                     4.25.0
jsonschema-specifications      2025.4.1
kaldi-native-fbank             1.22.2
keras                          3.11.3
keyring                        25.6.0
kiwisolver                     1.4.9
kornia                         0.8.1
kornia_rs                      0.1.9
LangSegment                    0.3.5
lark-parser                    0.12.0
lazy_loader                    0.4
libarchive-c                   5.3
libclang                       18.1.1
libmambapy                     2.1.1
librosa                        0.11.0
lief                           0.16.6
lightning-utilities            0.15.2
lintrunner                     0.12.7
litelama                       0.1.7
llvmlite                       0.45.0
loguru                         0.7.3
loralib                        0.1.2
lxml                           6.0.2
Mako                           1.3.10
manifold3d                     3.2.1
mapbox_earcut                  1.0.3
Markdown                       3.9
markdown-it-py                 4.0.0
MarkupSafe                     3.0.2
matplotlib                     3.10.6
matplotlib-inline              0.1.7
matrix-nio                     0.25.2
mdurl                          0.1.2
mediapipe                      0.10.14
menuinst                       2.3.1
ml_dtypes                      0.5.3
more-itertools                 10.8.0
mpmath                         1.3.0
msgpack                        1.1.1
mss                            10.1.0
multidict                      6.6.4
multiprocess                   0.70.16
namex                          0.1.0
natsort                        8.4.0
networkx                       3.5
ninja                          1.11.1.4
nodeenv                        1.9.1
numba                          0.62.0
numpy                          1.26.4
nvidia-cublas-cu12             12.8.4.1
nvidia-cuda-cupti-cu12         12.8.90
nvidia-cuda-nvrtc-cu12         12.8.93
nvidia-cuda-runtime-cu12       12.8.90
nvidia-cudnn-cu12              9.10.2.21
nvidia-cufft-cu12              11.3.3.83
nvidia-cufile-cu12             1.13.1.3
nvidia-curand-cu12             10.3.9.90
nvidia-cusolver-cu12           11.7.3.90
nvidia-cusparse-cu12           12.5.8.93
nvidia-cusparselt-cu12         0.7.1
nvidia-nccl-cu12               2.27.3
nvidia-nvjitlink-cu12          12.8.93
nvidia-nvtx-cu12               12.8.90
omegaconf                      2.3.0
onnx                           1.19.0
onnxruntime                    1.23.0
onnxruntime-gpu                1.23.0
open_clip_torch                3.2.0
openai                         1.109.1
openai-whisper                 20250625
opencv-contrib-python          4.12.0.88
opencv-contrib-python-headless 4.12.0.88
opencv-python                  4.12.0.88
opencv-python-headless         4.12.0.88
opt_einsum                     3.4.0
optree                         0.17.0
orjson                         3.11.3
packaging                      25.0
pandas                         2.3.2
parso                          0.8.4
pbs-installer                  2025.9.18
peft                           0.13.2
pexpect                        4.9.0
pickleshare                    0.7.5
piexif                         1.1.3
pilgram                        1.2.1
pillow                         11.0.0
pip                            25.1.1
pixeloe                        0.1.4
pkginfo                        1.12.1.2
platformdirs                   4.3.8
pluggy                         1.6.0
poetry                         2.2.1
poetry-core                    2.2.1
pooch                          1.8.2
portalocker                    3.2.0
pre_commit                     4.3.0
prettytable                    3.16.0
proces                         0.1.7
prompt_toolkit                 3.0.51
propcache                      0.3.2
protobuf                       6.33.0
psutil                         7.0.0
ptyprocess                     0.7.0
pure_eval                      0.2.3
py3langid                      0.3.0
pyarrow                        21.0.0
pybase16384                    0.3.8
pycairo                        1.28.0
pycocoevalcap                  1.2
pycocotools                    2.0.10
pycollada                      0.9.2
pycosat                        0.6.6
pycparser                      2.22
pycryptodome                   3.23.0
pydantic                       2.11.9
pydantic_core                  2.33.2
pydantic-settings              2.10.1
pydub                          0.25.1
PyGithub                       2.8.1
Pygments                       2.19.2
PyJWT                          2.8.0
pyloudnorm                     0.1.1
PyMatting                      1.1.14
pymediainfo                    7.0.1
PyNaCl                         1.6.0
pynini                         2.1.6
pyOpenSSL                      25.3.0
pyparsing                      3.2.5
pypinyin                       0.55.0
pyproject_hooks                1.2.0
PySocks                        1.7.1
pytest                         8.4.2
python-dateutil                2.9.0.post0
python-dotenv                  1.1.1
python-etcd                    0.4.5
python-multipart               0.0.20
python-socks                   2.7.2
pytorch-lightning              2.5.5
pytz                           2025.2
PyYAML                         6.0.2
pyzbar                         0.1.9
qrcode                         8.2
RapidFuzz                      3.14.1
redis                          6.4.0
referencing                    0.36.2
regex                          2025.9.18
rembg                          2.0.67
reportlab                      4.4.4
requests                       2.32.4
requests-toolbelt              1.0.0
rich                           14.1.0
rlPyCairo                      0.4.0
rpds-py                        0.26.0
rtree                          1.4.1
ruamel.yaml                    0.18.14
ruamel.yaml.clib               0.2.8
ruff                           0.14.0
s3transfer                     0.14.0
safehttpx                      0.1.6
safetensors                    0.6.2
sageattention                  1.0.6
SAM-2                          1.0
scenedetect                    0.6.7.1
scikit-image                   0.25.2
scikit-learn                   1.7.2
scipy                          1.16.2
SecretStorage                  3.4.0
segment-anything               1.0
semantic-version               2.10.0
sensevoice-onnx                1.1.0
sentencepiece                  0.2.0
setuptools                     65.0.0
shapely                        2.1.2
shellingham                    1.5.4
simple-lama-inpainting         0.1.2
simpleeval                     1.0.3
simsimd                        6.5.3
six                            1.17.0
smmap                          5.0.2
sniffio                        1.3.1
sortedcontainers               2.4.0
sounddevice                    0.5.2
soundfile                      0.13.1
soupsieve                      2.7
soxr                           1.0.0
spandrel                       0.4.1
SQLAlchemy                     2.0.43
stack_data                     0.6.3
starlette                      0.48.0
stringzilla                    4.0.14
svg.path                       7.0
svglib                         1.6.0
swarm                          0.1.0
sympy                          1.14.0
tabulate                       0.9.0
tenacity                       9.1.2
tensorboard                    2.20.0
tensorboard-data-server        0.7.2
tensorflow                     2.20.0
termcolor                      3.1.0
threadpoolctl                  3.6.0
tifffile                       2025.9.20
tiktoken                       0.12.0
timm                           1.0.20
tinycss2                       1.4.0
tokenizers                     0.19.1
tomesd                         0.1.3
toml                           0.10.2
tomlkit                        0.13.3
torch                          2.8.0+cu128
torchaudio                     2.8.0+cu128
torchdiffeq                    0.2.5
torchelastic                   0.2.2
torchmetrics                   1.8.2
torchsde                       0.2.6
torchvision                    0.23.0+cu128
tqdm                           4.67.1
traitlets                      5.14.3
trampoline                     0.1.2
transformers                   4.44.0
transparent-background         1.3.4
trimesh                        4.8.3
triton                         3.4.0
trove-classifiers              2025.9.11.17
truststore                     0.10.1
typeguard                      4.4.4
typer                          0.19.2
types-dataclasses              0.6.6
typing_extensions              4.14.1
typing-inspection              0.4.1
tzdata                         2025.2
uiautomation                   2.0.29
Unidecode                      1.4.0
unpaddedbase64                 2.1.0
urllib3                        2.5.0
uv                             0.9.3
uvicorn                        0.37.0
vector-quantize-pytorch        1.23.2
vhacdx                         0.0.8.post2
virtualenv                     20.34.0
vocos                          0.1.0
watchdog                       6.0.0
wcwidth                        0.2.13
webencodings                   0.5.1
websocket-client               1.8.0
websockets                     15.0.1
Werkzeug                       3.1.3
WeTextProcessing               1.0.4.1
wget                           3.2
wheel                          0.45.1
whisper-timestamped            1.15.9
wrapt                          1.17.3
xformers                       0.0.32.post2
xxhash                         3.6.0
yacs                           0.1.8
yapf                           0.43.0
yarl                           1.20.1
zhipuai                        2.1.5.20250825
zipp                           3.23.0
zstandard                      0.23.0
```

**Comment by finger92** [comment]
try to install transformers==4.44.0. i think this is the key point

**Comment by zhangbiwen** [comment]
> try to install transformers==4.44.0. i think this is the key point

i upgrade to  transformers==4.45 ,run  error,
